doc/connectivity/bluetooth/bluetooth-shell.rst

```diff
@@ -0,0 +1,357 @@
+
+Extended Advertising
+====================
+
+Let's now have a look at some extended advertising features, to enable extended advertising, use the
+`ext-adv` parameter.
+
+.. code-block:: console
+
+        uart:~$ bt adv-create conn-nscan ext-adv name-ad
+        Created adv id: 0, adv: 0x200022f0
+        uart:~$ bt adv-start
+        Advertiser[0] 0x200022f0 set started
+
+This will create an advertiser, connectable and non scannable.
+
+Filter Accept List
+******************
+
+It's possible to create a list of allowed addresses. That list can be used to
+connect to those addresses automatically, here is how to do it:
+
+.. code-block:: console
+
+        uart:~$ bt fal-add 47:38:76:EA:29:36 random
+        uart:~$ bt fal-add 66:C8:80:2A:05:73 random
+        uart:~$ bt fal-connect on
+
+It will then connect to the first available device. In the example, if both devices are advertising
+at the same time, we will connect to the first address added to the list.
+
+The Filter Accept List can also be used for scanning or advertising by using the option :code:`fal`.
+For example, if we want to scan for a bunch of selected addresses, we can set up a Filter Accept
+List, like that:
+
+.. code-block:: console
+
+        uart:~$ bt fal-add 65:4B:9E:83:AF:73 random
+        uart:~$ bt fal-add 73:72:82:B4:8F:B9 random
+        uart:~$ bt fal-add 5D:85:50:1C:72:64 random
+        uart:~$ bt scan on fal
+
+You should see only those three addresses reported by the scanner.
+
+Enabling security
+*****************
+
+When connected to a device, you can enable multiple level of security, here is the list for
+Bluetooth LE:
+
+* **1** No encryption and no authentication;
+* **2** Encryption and no authentication;
+* **3** Encryption and authentication;
+* **4** Bluetooth LE Secure Connection.
+
+To enable security, do :code:`bt security <level>`. For levels requiring authentication (level 3 and
+above), you must first set the authentication method, to do it, you can use :code:`bt auth all`.
+After that, when you will set the security level, you will be asked to confirm the passkey on both
+devices. On the shell side, do it with the command :code:`bt auth-passkey-confirm`.
+
+Pairing
+=======
+
+Enabling authentication requires the devices to be bondable. By default the shell is bondable. You
+can make the shell not bondable using :code:`bt bondable off`. You can list all the device you are
+paired with :code:`bt bonds`.
+
+The shell has a maximum of paired devices defined by :kconfig:option:`CONFIG_BT_MAX_PAIRED`. You can
+remove a paired device using :code:`bt clear <address: XX:XX:XX:XX:XX:XX> <type: (public|random)>`
+or simply remove all paired device with :code:`bt clear all`.
+
+GATT
+****
+
+On the following examples we will assume that you have two devices already connected.
+
+To perform services discovery on the client side, use :code:`gatt discover`. This should show you
```

Comment:
service discovery
This should print all the services that are available on the GATT server.